features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64SerializerTest.kt

```diff
@@ -654,6 +654,55 @@ internal class Base64SerializerTest {
         verify(mockSecondCallback).onReady()
     }
 
+    @Test
+    fun `M failover to manual bitmap creation W handleBitmap { bitmapDrawable returned empty bytearray }`(
+        @Mock mockCreatedBitmap: Bitmap
+    ) {
+        // Given
+        whenever(mockBitmapDrawable.bitmap).thenReturn(mockBitmap)
+        whenever(mockBitmap.width).thenReturn(fakeBitmapWidth)
+        whenever(mockBitmap.height).thenReturn(fakeBitmapHeight)
+
+        whenever(mockBitmap.isRecycled)
+            .thenReturn(true)
+            .thenReturn(false)
+
+        val emptyByteArray = ByteArray(0)
+        whenever(mockWebPImageCompression.compressBitmap(any()))
+            .thenReturn(emptyByteArray)
+            .thenReturn(fakeByteArray)
+
+        whenever(mockDrawableUtils.createScaledBitmap(any(), anyOrNull()))
+            .thenReturn(mockBitmap)
+            .thenReturn(mockCreatedBitmap)
+
+        whenever(mockBase64Utils.serializeToBase64String(any()))
+            .thenReturn(fakeBase64String)
+
+        // When
+        testedBase64Serializer.handleBitmap(
+            applicationContext = mockApplicationContext,
+            displayMetrics = mockDisplayMetrics,
+            drawable = mockBitmapDrawable,
+            drawableWidth = fakeBitmapWidth,
+            drawableHeight = fakeBitmapHeight,
+            imageWireframe = fakeImageWireframe,
+            base64SerializerCallback = mockSerializerCallback
+        )
+
+        // Then
+        verify(mockDrawableUtils, times(1)).createBitmapOfApproxSizeFromDrawable(
+            drawable = any(),
+            drawableWidth = any(),
+            drawableHeight = any(),
+            displayMetrics = any(),
+            requestedSizeInBytes = anyOrNull(),
+            config = anyOrNull(),
+            base64SerializerCallback = any(),
+            bitmapCreationCallback = any()
```

Comment:
is it possible to provide parameters we know for sure instead of just generic `any`?

Reply:
I don't know the value of the last callback since it's instantiated when DrawableUtils is called. However, I can use argumentCaptors and assert the values afterwards

Reply:
yes, I mean some parameters we probably already have, like mock of drawable, bitmaps, etc., so that we can use them instead  of `any`.